crates/pallets/starknet/src/tests/mod.rs

```diff
@@ -78,6 +79,41 @@ pub fn get_invoke_dummy(
     blockifier::transaction::transactions::InvokeTransaction { tx, tx_hash, only_query: false }
 }
 
+pub fn get_invoke_v3_dummy(
+    chain_id: Felt252Wrapper,
+    nonce: Nonce,
+) -> blockifier::transaction::transactions::InvokeTransaction {
+    let signature = TransactionSignature(vec![
+        StarkFelt::try_from("0x00f513fe663ffefb9ad30058bb2d2f7477022b149a0c02fb63072468d3406168").unwrap(),
+        StarkFelt::try_from("0x02e29e92544d31c03e89ecb2005941c88c28b4803a3647a7834afda12c77f096").unwrap(),
+    ]);
+    let sender_address =
+        ContractAddress(PatriciaKey(StarkFelt::try_from(constants::BLOCKIFIER_ACCOUNT_ADDRESS).unwrap()));
+    let calldata = Calldata(Arc::new(vec![
+        StarkFelt::try_from("0x024d1e355f6b9d27a5a420c8f4b50cea9154a8e34ad30fc39d7c98d3c177d0d7").unwrap(), /* contract_address */
+        StarkFelt::try_from("0x00e7def693d16806ca2a2f398d8de5951344663ba77f340ed7a958da731872fc").unwrap(), /* selector for the `with_arg` external */
+        StarkFelt::try_from("0x1").unwrap(),  // calldata_len
+        StarkFelt::try_from("0x19").unwrap(), // calldata[0]
+    ]));
+
+    let tx = starknet_api::transaction::InvokeTransaction::V3(starknet_api::transaction::InvokeTransactionV3 {
+        resource_bounds: create_resource_bounds(),
+        tip: starknet_api::transaction::Tip::default(),
+        calldata,
+        sender_address,
+        nonce,
+        signature,
+        nonce_data_availability_mode: DataAvailabilityMode::L1,
+        fee_data_availability_mode: DataAvailabilityMode::L1,
+        paymaster_data: starknet_api::transaction::PaymasterData(vec![StarkFelt::ZERO]),
```

Comment:
https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/transactions/#v3_transaction_fields

The spes says it should be empty

Reply:
same for `account_deployment_data`

Reply:
and `paymaster_data`

Reply:
It is set to zero but it's typed like it is in `starknet-api`. You can check it here https://github.com/starkware-libs/starknet-api/blob/77ff5a30e924cfe7672c773b140fe16af47a07b1/src/external_transaction_test.rs#L57

Reply:
What I meant is that the value should be `starknet_api::transaction::PaymasterData(vec![])`